`error: use of undeclared identifier` during `v run`

I'm still very new to this language. So much so, that this error happened when I was trying to do [Project Euler Problem 2](https://projecteuler.net/problem=2) using V.

Chances are that this is my fault, but I thought I'd open a ticket just in case. Thank you for maintaining this amazing language. 👍

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: macos, macOS, 12.2.1, 21D62
Processor: 8 cpus, 64bit, little endian, Apple M1 Pro
CC version: Apple clang version 13.1.6 (clang-1316.0.21.2)

getwd: /Users/harensamarasinghe/vlang
vmodules: /Users/harensamarasinghe/.vmodules
vroot: /opt/local/lib/vlang
vexe: /opt/local/lib/vlang/v
vexe mtime: 2022-03-30 14:37:18
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 509367b

Git version: git version 2.35.1
Git vroot status: Error: fatal: not a git repository (or any of the parent directories): .git
.git/config present: false
thirdparty/tcc: N/A

```

**What did you do?**
`v run hello.v`
```v
mut small := 1
mut big := 2
mut sum := 2

for big < 4_000_000 {
	small, big = big, small + big
	if big % 2 == 0 {
		sum += big
	}
}
println(big)
```

**What did you expect to see?**

A single integer being printed, representing the sum of all even fibonacci numbers up to 4 million.


**What did you see instead?**
```
hello.v:3:5: warning: unused variable: `sum`
    1 | mut small := 1
    2 | mut big := 2
    3 | mut sum := 2
      |     ~~~
    4 | 
    5 | for big < 4_000_000 {
==================
   ^~~~~~~~~~~~~~~
/opt/local/var/run/vlang/hello.17900573575609603610.tmp.c:11331:17: error: use of undeclared identifier 'small'
                int _var_65 = small;
                              ^
1 warning and 1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

Interestingly, this doesn't happen if I run `v hello.v`, so I'm not sure what's going on here:

```
./hello.v:1:1: error: bad top level statement keyword `mut`
    1 | mut small := 1
      | ~~~
    2 | mut big := 2
    3 | mut sum := 2
```

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -584,10 +584,10 @@ fn (mut g Gen) gen_cross_var_assign(node &ast.AssignStmt) {
 				left_sym := g.table.sym(left_typ)
 				if left_sym.kind == .function {
 					g.write_fn_ptr_decl(left_sym.info as ast.FnType, '_var_$left.pos.pos')
-					g.writeln(' = $left.name;')
+					g.writeln(' = ${c_name(left.name)};')
 				} else {
 					styp := g.typ(left_typ)
-					g.writeln('$styp _var_$left.pos.pos = $left.name;')
+					g.writeln('$styp _var_$left.pos.pos = ${c_name(left.name)};')
 				}
 			}
 			ast.IndexExpr {

--- a/vlib/v/tests/cross_assign_test.v
+++ b/vlib/v/tests/cross_assign_test.v
@@ -168,3 +168,24 @@ fn test_cross_assign_of_big_int() {
 	println(a)
 	assert a == big.one_int
 }
+
+fn test_cross_assign_of_reserved_name_variable() {
+	mut small := 1
+	mut big := 2
+	mut sum := 2
+
+	for big < 4_000_000 {
+		small, big = big, small + big
+		if big % 2 == 0 {
+			sum += big
+		}
+	}
+	println(small)
+	assert small == 3524578
+
+	println(big)
+	assert big == 5702887
+
+	println(sum)
+	assert sum == 4613732
+}

